File: Day1/Examples/C/broadcast.c
/*
 *
 * Rank 0 reads into a buffer and does the broadcast
 *
*/
#include <stdio.h>
#include <mpi.h>

int main( int argc, char **argv ) {
  int myRank, uniSize, ierror;
  int buf;

  ierror=MPI_Init(&argc, &argv);
  ierror=MPI_Comm_size(MPI_COMM_WORLD, &uniSize);
  ierror=MPI_Comm_rank(MPI_COMM_WORLD, &myRank);
  
  if ( myRank == 0 ) {
      printf( "Please insert a number\n" );
      scanf( "%d", &buf );
  }
  ierror = MPI_Bcast( &buf, 1, MPI_INT, 0, MPI_COMM_WORLD );
  printf( "I am process %d of %d and buf is %d\n", myRank, uniSize, buf );

  ierror = MPI_Finalize();
  return ierror;
}

File: Day1/Examples/Fortran/broadcast.f90
!
! Rank 0 reads in to a buffer and does the broadcast
!
program broadcast
    use mpi
    implicit none
    integer (kind=4) :: ierror, world_rank, world_size, buf

    call MPI_Init( ierror )
    if (ierror .ne. 0) stop ' Cannot initialize MPI '
    call MPI_Comm_rank( MPI_COMM_WORLD, world_rank, ierror )
    call MPI_Comm_size( MPI_COMM_WORLD, world_size, ierror )

    if ( world_rank .eq. 0 ) then
        write(6,*) ' Please insert a number'
        read(5,*) buf
    endif
    call MPI_Bcast( buf, 1, MPI_INTEGER, 0, MPI_COMM_WORLD, ierror )
    if (ierror .ne. 0) write(6,*) ' Problem with broadcast '
    write(6,'(a,i0,a,i0,a,i0)') 'I am process ',world_rank,' of ',world_size,' and buf is ',buf

    call MPI_Finalize(ierror);
end program broadcast
